src/core/worker.js

```diff
@@ -671,11 +671,9 @@ var WorkerMessageHandler = {
 
       var onFailure = function(e) {
         if (e instanceof PasswordException) {
-          if (e.code === PasswordResponses.NEED_PASSWORD) {
-            handler.send('NeedPassword', e);
-          } else if (e.code === PasswordResponses.INCORRECT_PASSWORD) {
-            handler.send('IncorrectPassword', e);
-          }
+          handler.sendWithPromise('Password', e).then(function (data) {
+            pdfManager.updatePassword(data.password);
+          });
```

Comment:
The code at pdfManagerReady below looks non-symmetrical to recovery password handling. I think we shall remove pdfManager.passwordChanged/._passwordChangedCapability? The pdfManagerReady call will be called after password update (we need to move pdfManagerReady from being anonymous function to closure before getPdfManager() call).